monty-31/src/dft/backward.rs

```diff
@@ -8,33 +8,206 @@ extern crate alloc;
 use alloc::vec::Vec;
 
 use itertools::izip;
+use p3_field::{Field, PackedFieldPow2, PackedValue};
+use p3_util::log2_strict_usize;
 
 use crate::utils::monty_reduce;
-use crate::{MontyField31, MontyParameters, TwoAdicData};
+use crate::{FieldParameters, MontyField31, TwoAdicData};
+
+#[inline(always)]
+fn backward_butterfly<T: PackedFieldPow2>(x: T, y: T, roots: T) -> (T, T) {
+    let t = y * roots;
+    (x + t, x - t)
+}
+
+#[inline(always)]
+fn backward_butterfly_interleaved<const HALF_RADIX: usize, T: PackedFieldPow2>(
+    x: T,
+    y: T,
+    roots: T,
+) -> (T, T) {
+    let (x, y) = x.interleave(y, HALF_RADIX);
+    let (x, y) = backward_butterfly(x, y, roots);
+    x.interleave(y, HALF_RADIX)
+}
+
+#[inline]
+fn backward_iterative_layer_0<T: PackedFieldPow2>(input: &mut [T], roots: &[T::Scalar]) {
```

Comment:
Hmm a slightly better name for this might be `backward_pass_packed`? Particularly with how it is regularly called by `backward_pass` whenever `backward_pass` can be done in a packed manner.